src/feasibility_form_nls.jl

```diff
@@ -239,10 +239,10 @@ function hess_coord!(nlp :: FeasibilityFormNLS, xr :: AbstractVector, rows :: Ab
   y2 = @view y[ne+1:ne+m]
   I = 1:nnzhF
   @views hess_coord_residual!(nlp.internal, x, y1, rows[I], cols[I], vals[I])
-  if m > 0
-    I = nnzhF+1:nnzhF+nnzhc
-    @views hess_coord!(nlp.internal, x, rows[I], cols[I], vals[I], obj_weight=0.0, y=y2)
-  end
+  I = nnzhF+1:nnzhF+nnzhc
+  @views hess_coord!(nlp.internal, x, rows[I], cols[I], vals[I], obj_weight=0.0, y=y2)
+  @views rows[nnzhF + nnzhc + 1 : nnzhF + nnzhc + ne] = n - ne + 1 : n  # Hessian of the residual ¹/₂‖r‖²
+  @views cols[nnzhF + nnzhc + 1 : nnzhF + nnzhc + ne] = n - ne + 1 : n
```

Comment:
We don't change `rows` and `cols`.

Below, line 246, should be the Hessian of `¹/₂‖r‖²` times `obj_weight`.

Reply:
A problem that may happen here is that for `NLS` problems some `hess` functions may be undefined due to the `J'*J` term. So calling `hess_coord!` here may lead to a `NotImplementedError`.